src/tests/suite/daos_nvme_recovery.c

```diff
@@ -142,6 +154,28 @@ nvme_recov_1(void **state)
 	set_fail_loc(arg, rank, fail_loc_tgt,
 		     DAOS_NVME_FAULTY | DAOS_FAIL_ALWAYS);
 
+	if (mode == 0) {
+		/**
+		*  Continue to check blobstore until state is "OUT"
+		*  or max test retry count is hit (5 min).
+		*/
+		rc = wait_and_verify_blobstore_state(
+			devices[faulty_disk_idx].device_id,
+			/*expected state*/"out", arg->group);
+		assert_rc_equal(rc, 0);
+
+		/**
+		* Connect the pool for query check.
+		 */
+		print_message("Connect the pool to get the pool query\n");
+		rc = daos_pool_connect(arg->pool.pool_uuid, arg->group,
+				       DAOS_PC_RW, &arg->pool.poh,
+				       &arg->pool.pool_info, NULL /* ev */);
+		assert_rc_equal(rc, 0);
+		/* Set container handle as invalid so it does not close again*/
+		arg->coh = DAOS_HDL_INVAL;
+	}
+
```

Comment:
@ravalsam does this work when testing locally? (ie just running the offline test on its own?)
So essentially in this test for offline, you are closing the single container for the pool, and then disconnecting the pool (taking it offline) before the faulty device reaction is simulated, and then trying to reconnect the pool to continue the rest of the test? This might be a valid test case, however I was thinking that offline test would instead involve running the online faulty reaction test first, and then disconnecting the pool, and then creating a new pool/cont and running the same test again (but this time with an offline pool present in the system). If I remember correctly this is where we were running into errors before and we needed a test case for this.

@niu any thoughts on this?

Reply:
Yes, only offline test its own works fine. 
So when we inject the fault, the pool is offline in this case. But let me know if we want to test scenarios where we need online and offline in the same test sequence and have fault injected for both conditions.

Reply:
I'm fine with current way, maybe we can add more tests with multiple pools (online & offline mixed) in the future.

Reply:
Sounds like a plan!

Reply:
Yes, I created a new Ticket DAOS-7120 to add a mixed pool during NVMe fault and will add a new test in a separate PR.